dvl-a50/dvl.py

```diff
@@ -175,12 +194,44 @@ def set_should_send(self, should_send):
         self.should_send = should_send
         self.save_settings()
 
+    def longitude_scale(self, lat):
+        scale = math.cos(lat * (math.pi / 180.0))
+        return max(scale, 0.01)
+
+    def latLngToXY (self, lat, lon):
+        x = (lat-self.origin[0]) * LATLON_TO_CM
+        y =  self.longitude_scale((lat + self.origin[0])/2) * LATLON_TO_CM * (lon-self.origin[1])
+        return [x, y]
+
+    def has_origin_set(self):
+        data = json.loads(self.mav.get("/GLOBAL_POSITION_INT/message"))
+        return data['lat'] != 0 or data['lon'] != 0
+
+    def set_current_position(self, lat, lon):
+        """
+        Sets the EKF origin to lat, lon
+        """
+        lat = float(lat)
+        lon = float(lon)
+        if not self.has_origin_set():
+            self.mav.set_gps_origin(lat, lon)
+            self.origin = [lat, lon]
+            self.save_settings()
+        else:
+            x, y = self.latLngToXY (lat, lon)
+            depth = float(self.mav.get("/VFR_HUD/message/alt"))
+
+            attitude = json.loads(self.mav.get("/ATTITUDE"))
+            positions = [x, y, -depth]
+            self.reset_counter += 1
+            self.mav.send_vision_position_estimate(self.timestamp, positions, reset_counter=self.reset_counter)
+
     def set_gps_origin(self, lat, lon):
         """
         Sets the EKF origin to lat, lon
         """
         self.mav.set_gps_origin(lat, lon)
-        self.origin = [lat, lon]
+        self.origin = [float(lat), float(lon)]
```

Comment:
Is this conversion necessary?
- If so, should it happen before the values get sent to over MAVLink? 
- If not, should the function signature be typed to specify floats as the expected input?